### PR TITLE
new copy/paste mechanism

### PR DESCRIPTION
**Pull request by vxsx**
little bit of name refactoring + more unit tests
also cherry-picks https://github.com/divio/django-cms/pull/5199/commits/d8375635c392911483c16b4c4697222230298ea7 since there's no guarantee that #5199 will make it into the release

before:
paste button directly on the node
screen shot 2016-04-29 at 09 11 36
screen shot 2016-04-29 at 09 11 38

after:
paste buttons in the menus

screen shot 2016-04-29 at 09 11 15
screen shot 2016-04-29 at 09 11 19

### CODE REVIEW

**Comment by Brankub** [comment]
Overall looks good. Huge changes 👍

File: cms/static/cms/js/modules/cms.pagetree.js [hunk 1]
```diff
@@ -359,17 +372,18 @@ var CMS = window.CMS || {};
...
-                this.cache.type = null;
-                this._hideHelpers();
+            if (this.clipboard.type === obj.type && jsTreeId === this.clipboard.id) {
+                this.clipboard.type = null;
+                this.clipboard.id = null;
+                this.clipboard.origin = null;
+                this._disablePaste();
             } else {
-                // we need to cache the node and type so `_showHelpers`
+                // we need to cache the node in the "clipboard" the node and type so `_enablePaste`
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.pagetree.js]: I don't get the comment ;)

File: cms/static/cms/sass/components/pagetree/_tree.scss [hunk 2]
```diff
@@ -46,6 +46,36 @@
     .cms-hover-tooltip {
         z-index: auto;
     }
+    .sr-only {
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/pagetree/_tree.scss]: shouldn't we add this as a helper class in general for the cms? like cms-sr-only?

**Comment by FinalAngel** [comment]
nice work 👍

File: cms/static/cms/sass/components/pagetree/_tree.scss (hunk 2 above)
Comment (vxsx) [inline cms/static/cms/sass/components/pagetree/_tree.scss]: i was thinking about that, not in this PR

Comment (FinalAngel) [inline cms/static/cms/sass/components/pagetree/_tree.scss]: 👍

**Pull request merged**